nemo/collections/nlp/data/machine_translation/sequence_to_sequence_dataset.py

```diff
@@ -0,0 +1,93 @@
+# Copyright (c) 2022, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+
+import numpy as np
+import torch
+
+from nemo.collections.common.tokenizers.tokenizer_spec import TokenizerSpec
+from nemo.collections.nlp.data.language_modeling.text_memmap_dataset import TextMemMapDataset
+from nemo.core.classes import Dataset
+from nemo.utils import logging
+
+
+class SequenceToSequenceDataset(Dataset):
+    """Sequence to Sequence Dataset in memory."""
+
+    def __init__(
+        self,
+        src_file_name: str,
+        tgt_file_name: str,
+        src_tokenizer: TokenizerSpec,
+        tgt_tokenizer: TokenizerSpec,
+        max_src_seq_length: int,
+        max_tgt_seq_length: int,
+    ):
+        self.src_file_name = src_file_name
+        self.tgt_file_name = tgt_file_name
+        self.src_tokenizer = src_tokenizer
+        self.tgt_tokenizer = tgt_tokenizer
+        self.max_src_seq_length = max_src_seq_length
+        self.max_tgt_seq_length = max_tgt_seq_length
+        assert self.max_src_seq_length > 0
+        assert self.max_tgt_seq_length > 0
+
+        self.src_dataset = TextMemMapDataset(dataset_paths=[src_file_name], tokenizer=src_tokenizer)
+        self.tgt_dataset = TextMemMapDataset(dataset_paths=[tgt_file_name], tokenizer=tgt_tokenizer)
+        assert len(self.src_dataset) == len(self.tgt_dataset), "src and tgt has different number of lines"
+
+    def __len__(self):
+        return len(self.examples)
+
+    def __getitem__(self, idx):
+        example = self.examples[idx]
```

Comment:
`self.examples` is not used here and was there in the old not memmapped seq2seq dataset class. We should be removing this no?

Reply:
Fixed